urllib3/connectionpool.py

```diff
@@ -239,11 +239,15 @@ def _get_conn(self, timeout=None):
         # If this is a persistent connection, check if it got disconnected
         if conn and is_connection_dropped(conn):
             log.info("Resetting dropped connection: %s" % self.host)
-            conn.close()
-            if getattr(conn, 'auto_open', 1) == 0:
-                # This is a proxied connection that has been mutated by
-                # httplib._tunnel() and cannot be reused (since it would
-                # attempt to bypass the proxy)
+            try:
+                conn.close()
+                if getattr(conn, 'auto_open', 1) == 0:
+                    # This is a proxied connection that has been mutated by
+                    # httplib._tunnel() and cannot be reused (since it would
+                    # attempt to bypass the proxy)
+                    conn = None
+            except Exception as e:
```

Comment:
~~~
>>> import inspect
>>> inspect.getmro(OpenSSL.SSL.SysCallError)
(<class 'OpenSSL.SSL.SysCallError'>,
 <class 'OpenSSL.SSL.Error'>,
 <type 'exceptions.Exception'>,
 <type 'exceptions.BaseException'>,
 <type 'object'>)
~~~

Unfortunately not a child of `EnvironmentError`, but that might be a reasonable thing to catch as well, whether or not pyopenssl is available.